src/RA_AchievementSet.cpp

```diff
@@ -431,43 +389,47 @@ BOOL AchievementSet::FetchFromWebBlocking(ra::GameID nGameID)
 BOOL AchievementSet::LoadFromFile(ra::GameID nGameID)
 {
     if (nGameID == 0)
-    {
         return TRUE;
+
+    const auto sFilename{ GetAchievementSetFilename(nGameID) };
+    std::ifstream ifile{ sFilename };
+    if (!ifile.is_open())
+    {
+        //	Cannot open file
+        RA_LOG("Cannot open file %s\n", sFilename.c_str());
+        char sErrMsg[2048U]{};
+        strerror_s(sErrMsg, errno);
+        RA_LOG("Error: %s\n", sErrMsg);
+        return FALSE;
     }
 
-    const std::string sFilename = GetAchievementSetFilename(nGameID);
+    //	Store this: we are now assuming this is the correct checksum if we have a file for it
+    g_pCurrentGameData->SetGameID(nGameID);
 
-    FILE* pFile = nullptr;
-    errno_t nErr = fopen_s(&pFile, sFilename.c_str(), "r");
-    if (pFile != nullptr)
+    if (m_nSetType == Local)
     {
-        //	Store this: we are now assuming this is the correct checksum if we have a file for it
-        g_pCurrentGameData->SetGameID(nGameID);
-
-        if (m_nSetType == Local)
-        {
-            const char EndLine = '\n';
-            char buffer[4096];
-            unsigned long nCharsRead = 0;
+        auto buffer{ std::make_unique<char[]>(4096U) };
```

Comment:
I still think this one needs be dynamic because it's used by `getline` and the `while` loop below. A lot of the other ones I made them static again where it made sense.